esp/bindings/http/platform/httpservice.cpp

```diff
@@ -1549,6 +1563,9 @@ EspAuthState CEspHttpServer::checkUserAuthPerRequest(EspAuthRequest& authReq)
     ESPLOG(LogMax, "checkUserAuthPerRequest");
 
     authReq.authBinding->populateRequest(m_request.get());
+    IEspSecureContextEx* secureContextEx = dynamic_cast<IEspSecureContextEx*>(authReq.ctx->querySecureContext());
```

Comment:
I'm confident that the use of dynamic_cast will be unpopular. My quick check shows ~15 preexisting references to IEspSecureContext that could be changed to IEspSecureContextEx, and the two casts you've added would not be needed.